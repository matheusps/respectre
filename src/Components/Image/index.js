import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const Image = ({ src, responsive, fit, className, ...props }) => {
  const imageClasses = classNames(className, `img-fit-${fit}`, {
    'img-responsive': responsive,
  })
  return <img className={imageClasses} src={src} {...props} />
}

Image.propTypes = {
  /** Image source */
  src: PropTypes.string.isRequired,
  /** If is responsive or not */
  responsive: PropTypes.bool,
  /** Image fit relative to it's parent */
  fit: PropTypes.oneOf(['contain', 'cover']),
}

Image.defaultProps = {
  responsive: false,
}

export default Image
