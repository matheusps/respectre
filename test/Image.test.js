import React from 'react'
import Image from '../src/Components/Image'
import renderer from 'react-test-renderer'
import { classNames } from 'classnames'

describe('Image', () => {
  it('should match snapshot', () => {
    const component = renderer
      .create(<Image src="testing" responsive fit="contain" />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })

  it('should not have any children', () => {
    const img1 = renderer.create(<Image src="..." />).toJSON()
    const img2 = renderer.create(<Image src="..." responsive />).toJSON()
    const img3 = renderer.create(<Image src="..." fit="cover" />).toJSON()

    expect(img1.children).toBeNull
    expect(img2.children).toBeNull
    expect(img3.children).toBeNull
  })

  it('should have the correct props', () => {
    const img = renderer
      .create(<Image src="." responsive fit="cover" />)
      .toJSON()

    expect(img.props.src).toEqual('.')
    expect(img.props.className).toContain('img-responsive')
    expect(img.props.className).toContain('img-fit-cover')
  })
})
