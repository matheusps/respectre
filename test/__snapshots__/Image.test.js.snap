// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should match snapshot 1`] = `
<img
  className="img-fit-undefined"
  src="testing"
/>
`;

exports[`Image should match snapshot with props 1`] = `
<img
  className="img-fit-contain img-responsive"
  src="testing"
/>
`;
